### PR TITLE
default 16bit rgb/rgba textures to unorm instead of uint

### LINKED ISSUE #8185: Bevy 0.10 Res<AssetServer>::load(...) Causes Panic Given Valid Path
## Bevy version

0.10

## \[Optional\] Relevant system information

Rust version: cargo 1.68.1 (115f34552 2023-02-26)
OS: macOS Ventura 13.2.1
AdapterInfo { name: "Apple M1", vendor: 0, device: 0, device_type: IntegratedGpu, driver: "", driver_info: "", backend: Metal }

## What you did

I created a simple Bevy project and added an 'assets' directory that is a sibling of the 'src' directory. The structure is as follows:

- ROOT
    - assets
        - Idle.png
    - src
    - target
    

## What went wrong

A panic occurs when trying to fetch the "Idle.png" using ```asset_server.load("Idle.png")```. Unfortunately, I am unable to make sense of the stack trace and haven’t been able to find anything else about this issue online.

- what were you expecting?

I would expect the application to run (without crashing) and display my ```Idle.png``` (44x47) in the center of the screen.

- what actually happened?

panic occurs during ```startup_system```

## Additional information

Project code:

```
use bevy::prelude::*;

fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_startup_system(setup)
        .run();
}

fn setup(
    mut commands: Commands,
    asset_server: Res<AssetServer>,
) {
    
    commands.spawn(Camera2dBundle::default());

    commands.spawn(SpriteBundle {
        texture: asset_server.load("Idle.png"), // Causes panic
        transform: Transform::from_translation(Vec3::new(0., 0., 0.)),
        ..default()
    });
}
```

Log:
```
Caused by:
    In Device::create_bind_group
      note: label = `sprite_material_bind_group`
    texture binding 0 expects sample type = Float { filterable: true }, but given a view with format = Rgba16Uint

', /Users/aleksei/.cargo/registry/src/{github}/wgpu-0.15.1/src/backend/direct.rs:3024:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'Compute Task Pool (3)' panicked at 'A system has panicked so the executor cannot continue.: RecvError', /Users/aleksei/.cargo/registry/src/{github}/bevy_ecs-0.10.0/src/schedule/executor/multi_threaded.rs:194:60
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', /Users/aleksei/.cargo/registry/src/{github}/bevy_tasks-0.10.0/src/task_pool.rs:376:49
thread 'Compute Task Pool (3)' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError', /Users/aleksei/.cargo/registry/src/{github}/bevy_render-0.10.0/src/pipelined_rendering.rs:136:45
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /Users/aleksei/.cargo/registry/src/{github}/bevy_tasks-0.10.0/src/task_pool.rs:376:49
```

### PR DESCRIPTION
**Pull request by robtfm**
# Objective

fix  #8185, #6710
replace #7005 (closed)

rgb and rgba 16 bit textures currently default to `Rgba16Uint`, the more common use is `Rgba16Unorm`, which also matches the default type of rgb8 and rgba8 textures.

## Solution

Change default to `Rgba16Unorm`

### CODE REVIEW

**Review by superdump — APPROVED** [review]
Seems OK to me.

File: crates/bevy_render/src/texture/dds.rs [hunk 1]
```diff
@@ -88,7 +88,7 @@ pub fn dds_format_to_texture_format(
                     TextureFormat::Bc3RgbaUnorm
                 }
             }
-            D3DFormat::A16B16G16R16 => TextureFormat::Rgba16Uint,
+            D3DFormat::A16B16G16R16 => TextureFormat::Rgba16Unorm,
```
Comment (Elabajaba) [inline crates/bevy_render/src/texture/dds.rs:91]: Checked the microsoft docs and it seems like A16B16G16R16 is an old dx9 format and the dx11 equivalent is DXGI_FORMAT_R16G16B16A16_UNORM, so this seems correct.

File: crates/bevy_render/src/texture/image_texture_conversion.rs [hunk 2]
```diff
@@ -79,34 +79,20 @@ impl Image {
...
 
-                    local_data.extend_from_slice(&r.to_ne_bytes());
-                    local_data.extend_from_slice(&g.to_ne_bytes());
-                    local_data.extend_from_slice(&b.to_ne_bytes());
-                    local_data.extend_from_slice(&a.to_ne_bytes());
-                }
+                let raw_data = i.into_raw();
 
-                data = local_data;
+                data = cast_slice(&raw_data).to_owned();
```
Comment (Elabajaba) [inline crates/bevy_render/src/texture/image_texture_conversion.rs:90]: Should this potentially be in an unsafe block?

File: crates/bevy_render/src/texture/image_texture_conversion.rs [hunk 3]
```diff
@@ -79,34 +79,20 @@ impl Image {
...
-                    Vec::with_capacity(width as usize * height as usize * format.pixel_size());
-
-                for pixel in image.into_raw().chunks_exact(3) {
-                    // TODO: use the array_chunks method once stabilised
-                    // https://github.com/rust-lang/rust/issues/74985
-                    let r = pixel[0];
-                    let g = pixel[1];
-                    let b = pixel[2];
-                    let a = u16::max_value();
+            DynamicImage::ImageRgb16(i) => {
```
Comment (Elabajaba) [inline crates/bevy_render/src/texture/image_texture_conversion.rs:82]: I guess it matches the rest of the file, but I hate that we're using i as a variable name for something that isn't a loop index.

Comment (robtfm) [reply crates/bevy_render/src/texture/image_texture_conversion.rs:82]: agreed, changed

File: crates/bevy_render/src/texture/image_texture_conversion.rs (hunk 2 above)
Comment (robtfm) [reply crates/bevy_render/src/texture/image_texture_conversion.rs:90]: do you mean make `Image::from_dynamic` unsafe? i don't think it's necessary, the worst that'll happen is garbage pixeldata in your image.

**Review by mockersf — APPROVED** [review]

**Pull request merged**